servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -239,11 +240,18 @@ public void onNext(final ServiceDiscovererEvent<ResolvedAddress> event) {
                     return singletonList(createHost(addr));
                 }
 
-                if (handleExpired) {
-                    for (Host<ResolvedAddress, C> host : oldHostsTyped) {
-                        if (host.address.equals(addr) && host.tryToMarkActive()) {
-                            return oldHostsTyped;
+                // duplicates are not allowed
```

Comment:
do we need to actively prevent/detect duplicates unconditionally? the API currently doesn't allow for duplicates and our most popular discoverer (DNS) doesn't send duplicates. consider reverting to retain the `handleExpired` to avoid O(n) iteration before copy/add, and/or pre-allocate `newHosts` and add while iterating (duplicates are not expected/supported).